Source/Details/ASDataController.mm

```diff
@@ -629,6 +629,10 @@ - (void)updateWithChangeSet:(_ASHierarchyChangeSet *)changeSet
   Class<ASDataControllerLayoutDelegate> layoutDelegateClass = [self.layoutDelegate class];
   ++_editingTransactionGroupCount;
   dispatch_group_async(_editingTransactionGroup, _editingTransactionQueue, ^{
+    if (_deallocDelegate.isDeallocating) {
```

Comment:
Check to make sure `_deallocDelegate` isn't `nil` because if it is, `isDeallocating` will return `NO`. We really don't want to proceed in that case.

Reply:
Also, we should weakify & strongify `self`, and bail early if `self` is nil. It may not help with the crash, but it'll at least allow the data controller to be de-allocated much earlier (which is beneficial because it owns all of the cell nodes).

Reply:
Forgive me if I'm misunderstanding, but wouldn't _deallocDelegate itself be a sufficient indication of whether the object is deallocating? As a weak pointer, the contract is that the pointer should become nil before the dealloc method is entered. This is essential for objc4 to guarantee because a retain during dealloc is forbidden, and if a weak pointer were still available, a retain would be easily possible.

Reply:
hm I assumed it would still be useful since `_delegate` as a weak pointer references the same object, and (given the root cause assumption in this PR) `_delegate` can perform a valid `objcmsg_send`